packages/msgraph/src/Adaptor.js

```diff
@@ -76,14 +85,195 @@ export function create(resource, data, callback) {
+ *    - owner {string} - The type of drive owner (e.g. sites, groups).
+ * @param {string} name - The local name of the drive used to write to state.drives, ie, state.drives[name]
+ * @param {function} [callback = s => s] (Optional) Callback function
+ * @return {Operation}
+ */
+export function getDrive(specifier, name = 'default', callback = s => s) {
+  return state => {
+    const { accessToken, apiVersion } = state.configuration;
+    const [resolvedSpecifier, resolvedName] = expandReferences(
+      state,
+      specifier,
+      name
+    );
+
+    const { id, owner = 'drive' } = resolvedSpecifier;
+
+    let urlPath;
+    if (owner === 'drive') {
+      urlPath = `drives/${id}`;
+    } else {
+      urlPath = `${owner}/${id}/drive`;
+    }
+
+    const url = getUrl(urlPath, apiVersion);
+    const auth = getAuth(accessToken);
+
+    return request(url, { ...auth }).then(response => {
+      state.drives[resolvedName] = response;
+      return callback(state);
+    });
+  };
+}
+
+/**
+ * Get folder metadata or an array of folder contents.
+ * @public
+ * @example <caption>Get a folder by ID</caption>
+ * getFolder('01LUM6XOCKDTZKQC7AVZF2VMHE2I3O6OY3')
+ * @example <caption>Get a folder for a named drive by id</caption>
+ * getFolder("01LUM6XOCKDTZKQC7AVZF2VMHE2I3O6OY3",{ driveName: "mydrive"})
+ * @param {string} pathOrId - A path to a folder or folder id
+ * @param {object} options - (Optional) Query parametes
+ * @param {function} [callback = s => s] (Optional) Callback function
+ * @return {Operation}
+ */
+export function getFolder(pathOrId, options, callback = s => s) {
+  return async state => {
+    const defaultOptions = {
+      driveName: 'default', // Named drive in state.drives
+      metadata: false, // If false return folder files if true return folder metadata
+      $filter: '', // Eg: "file/mimeType eq \'application/vnd.ms-excel\'"
+    };
+    const { accessToken, apiVersion } = state.configuration;
+    const [resolvedPathOrId, resolvedOptions] = expandReferences(
+      state,
+      pathOrId,
+      options
+    );
+
+    const { driveName, metadata } = { ...defaultOptions, ...resolvedOptions };
+
+    if (!state.drives[driveName]) {
+      const egJobCode = [
+        'Eg: Get a site drive then get "/Sample Data" folder',
+        'getDrive({ id: "openfn.sharepoint.com", owner: "sites"})',
+        'getFolder("/Sample Data")',
+      ].join('\n\t  ');
+
+      const errorString = [
+        `Message: Drive is not defined`,
+        `Quick Fix: Add getDrive() operation before getFolder()`,
+        `${egJobCode}`,
+      ].join('\n\t∟ ');
+
+      throw new Error(errorString);
+    }
+
+    const { id: driveId } = state.drives[driveName];
+
+    let urlPath;
+
+    if (resolvedPathOrId.startsWith('/')) {
+      urlPath = `drives/${driveId}/root:/${encodeURIComponent(
+        resolvedPathOrId
+      )}`;
+      metadata ? urlPath : (urlPath = `${urlPath}:/children`);
```

Comment:
This ternary shoud be written as
```
if (metadata) {
   urlPath += "/:children";
}
```
Also it should be done after the if/else block, because it's the same code either way.

Check this pattern across this file.

Reply:
👍🏽 , i have added `if (!metadata) {
      urlPath += resolvedPathOrId.startsWith('/') ? ':/children' : '/children';
    }` after `if/else` block